File: 2015/10/solution.js
const solve = (input, times) => {
  let saying = input
  for (let i = 0; i < times; i++) {
    let next = ''
    let prevChar = saying.charAt(0)
    let count = 1
    for (let c = 1; c < saying.length; c++) {
      const currChar = saying.charAt(c)
      if (currChar === prevChar) {
        count += 1
      } else {
        next += `${count}${prevChar}`
        prevChar = currChar
        count = 1
      }
    }
    saying = next + `${count}${prevChar}`
  }
  return saying.length
}

const input = require('fs').readFileSync('input.txt', 'utf8')

console.log('Part 1', solve(input, 40))
console.log('Part 2', solve(input, 50))

File: 2015/10/solution.rb
def solve(input, times)
  saying = input
  for i in 0...times
    nxt = ''
    prev_char = saying[0]
    count = 1
    saying[1, saying.size - 1].each_char do |curr_char|
      if curr_char == prev_char
        count += 1
      else
        nxt << count.to_s + prev_char
        prev_char = curr_char
        count = 1
      end
    end
    nxt << count.to_s + prev_char
    saying = nxt
  end
  saying.size
end

input = File.read('input.txt')

puts 'Part 1', solve(input, 40)
puts 'Part 2', solve(input, 50)
